server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInAction.java

```diff
@@ -23,44 +23,61 @@
 import org.sonar.api.server.ws.Request;
 import org.sonar.api.server.ws.Response;
 import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.permission.OrganizationPermission;
 import org.sonar.server.qualityprofile.QProfileReset;
+import org.sonar.server.user.UserSession;
 
 import static org.sonar.server.util.LanguageParamUtils.getExampleValue;
 import static org.sonar.server.util.LanguageParamUtils.getLanguageKeys;
+import static org.sonarqube.ws.client.component.ComponentsWsParameters.PARAM_ORGANIZATION;
 
 public class RestoreBuiltInAction implements QProfileWsAction {
 
+  private final DbClient dbClient;
   private final QProfileReset reset;
   private final Languages languages;
+  private final UserSession userSession;
   private final QProfileWsSupport qProfileWsSupport;
 
-  public RestoreBuiltInAction(QProfileReset reset, Languages languages, QProfileWsSupport qProfileWsSupport) {
+  public RestoreBuiltInAction(DbClient dbClient, QProfileReset reset, Languages languages, UserSession userSession, QProfileWsSupport qProfileWsSupport) {
+    this.dbClient = dbClient;
     this.reset = reset;
     this.languages = languages;
+    this.userSession = userSession;
     this.qProfileWsSupport = qProfileWsSupport;
   }
 
   @Override
   public void define(WebService.NewController controller) {
-    WebService.NewAction restoreDefault = controller.createAction("restore_built_in")
+    WebService.NewAction action = controller.createAction("restore_built_in")
       .setDescription("Restore built-in profiles from the definitions declared by plugins. " +
         "Missing profiles are created, existing ones are updated.")
       .setSince("4.4")
       .setPost(true)
       .setHandler(this);
-    restoreDefault.createParam("language")
+    action.createParam("language")
       .setDescription("Restore the built-in profiles defined for this language")
       .setExampleValue(getExampleValue(languages))
       .setPossibleValues(getLanguageKeys(languages))
       .setRequired(true);
+
+    qProfileWsSupport.createOrganizationParam(action)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2209)